### DISCUSSION TITLE
Reference other table in permissions

### DISCUSSION THREAD

**Discussion by ptrxyz** [pocketbase/pocketbase#2493] [Q&A]
Imagine the following example:

I have a simple app allowing users to store a simple line of text:

Table Layout:
```
id: <autogenerated>
text: string
owner: reference to user
```

To only allow registered users to create stuff, the permissions would be something like:
```
@request.auth.id != ""
```

Now to my question: I would like to also limit the amount of texts a user is allowed to store. 
I thought this would be possible by creating a separate view that counts the texts a user has created:

```
SELECT texts.id, count(texts.id) as count, owner FROM texts;
```

And then adjust my permissions to 

```
@request.auth.id != "" && 
@collection.textCountingView.count < 5 && 
@collection.textCountingView.owner.id == @request.auth.id
```

with the intention of this creating something like a join. But this approach doesn't seem to work. Is there another way of getting this done without having to dig into the depth of using Go?

**Comment by ganigeorgiev** [comment]
You should be able to accomplish what you want with a helper view collection, but I'm not sure that I understand what the count is supposed to represent. If you want to count the number of records **per each owner** then the view query should be something like:

```sql
SELECT
    -- we use the owner as id as it would be unique but you can replace it with something else
    owner as id,
    owner,
    count(id) as count
FROM texts
GROUP BY owner
```

And then for the API rules you can use something like:

```js
@request.auth.id != "" && 
@collection.textCountingView.owner ?= @request.auth.id &&
@collection.textCountingView.count ?< 5
```

(_note that the `@collection.*` filter resolves to multiple records and therefore we use the `?` operator prefix to indicate that we want any/"at least one of" the `textCountView` records to satisfy the above requirements_)

**Reply by ptrxyz** [reply]
Ah, thanks @ganigeorgiev for the quick response. But is that really working? I mean, the 2nd condition is true, that as soon as I have at least one record where the owner ids match, and the third one as soon as there is any record in the list with count < 5.

Let's assume this:
```
// counting view:
// id, owner, count
1,  "Person1",  5;
2,  "Person2",  3;
3,  "Person3",  6;
```

If "Person1" wants to write something, the conditions resolve as follows:
```
@collection.textCountingView.owner ?= @request.auth.id &&    // true, "Person1" is in the counting view
@collection.textCountingView.count ?< 5                      // true, there is a row with count < 5 (namely row 2, owned by "Person2")
```

My understanding was (and that might totally be wrong), that Pocketbase would run the conditions one after the other on the all the rows. But from your answer I assume those conditions are more like a stacked filter? After condition 1 is applied, the 2nd condition is only applied to the result and not the whole view?

To explain what I mean, the difference for the example above would be:

Type 1: condition 1 runs on row1, row2, row3 and then condition 2 runs on row1, row2, row3.
Type 2: condition 1 runs on row1, row2, row3 and then condition 2 runs on row1 only, since it's the only one left after condition 1 was applied.

I suppose Type 2 is correct then?

**Reply by ganigeorgiev** [reply]
If you have multiple `@collection.textCountingView.*` conditions they should apply on a single row.
Internally `@collection` performs a regular join and the generated query would be something like:

```sql
select *
from texts
left join textCountingView tcv
where (
    {auth.id} != "" AND
    tcv.owner = {auth.id} AND
    tcv.count < 5
)
```

**Reply by ptrxyz** [reply]
Ok, great! I'll give it a try later on and mark the question as solved if it works. Thanks a bunch so far!

Edit: works like a charm! Thanks a bunch, consider this solved, can be closed.
